src/main/java/gregtech/api/recipes/RecipeMap.java

```diff
@@ -236,6 +236,31 @@ public Recipe findRecipe(long voltage, List<ItemStack> inputs, List<FluidStack>
         }
     }
 
+    /**
+     * Finds a Recipe matching ItemStack Inputs ignoring any required fluid input .
+     *
+     * @param voltage                 Voltage of the Machine or Long.MAX_VALUE if it has no Voltage
+     * @param inputs                  the Item Inputs
+     * @param outputFluidTankCapacity minimal capacity of output fluid tank, used for fluid canner recipes for example
+     * @return the Recipe it has found or null for no matching Recipe
+     */
+    @Nullable
+    public Recipe findRecipe(long voltage, List<ItemStack> inputs, int outputFluidTankCapacity) {
+        if (recipeList.isEmpty())
+            return null;
+        if (minInputs > 0 && GTUtility.amountOfNonEmptyStacks(inputs) < minInputs) {
+            return null;
+        }
+        if (inputs.isEmpty())
+            return null;
+        if (maxInputs > 0) {
+            return findByInputs(voltage, inputs);
+        } else {
+            return null;
+        }
+    }
```

Comment:
This is not good design as it is prone to errors. Because you have two overrides of function with different results which are not predictable and not explicitly defined by params and honestly I am not checking every override for documentation but only longest one. I think it would be better if you add override which takes enum value informing it how matching should work (vales for that matching enum should be default - both items and fluids, onlyItems, onlyFluids). There will be also need for some rearranging of functions calls so everything goes through longest override.